blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java

```diff
@@ -51,101 +46,57 @@
         return search(tenant, Arrays.asList(query));
     }
 
-
     public List<SearchResult> search(String tenant, List<String> queries) throws Exception {
-        String[] indexes = getIndexesToSearch();
-
-        return searchESByIndexes(tenant, queries, indexes);
+        return searchUsingRestApi(tenant, queries);
     }
 
-    private List<SearchResult> searchESByIndexes(String tenant, List<String> queries, String[] indexes) {
-        List<SearchResult> results = new ArrayList<SearchResult>();
+    private List<SearchResult> searchUsingRestApi(String tenant, List<String> queries) {
+        List<SearchResult> results = new ArrayList<>();
         Timer.Context multiSearchCtx = searchTimer.time();
-        SearchResponse response;
+        int hitsCount = 0;
+
         try {
             queryBatchHistogram.update(queries.size());
-            BoolQueryBuilder bqb = boolQuery();
-            QueryBuilder qb;
-
-            for (String query : queries) {
-                GlobPattern pattern = new GlobPattern(query);
-                if (!pattern.hasWildcard()) {
-                    qb = termQuery(ESFieldLabel.metric_name.name(), query);
-                } else {
-                    qb = regexpQuery(ESFieldLabel.metric_name.name(), pattern.compiled().toString());
-                }
-
-                bqb.should(boolQuery()
-                                .must(termQuery(ESFieldLabel.tenantId.toString(), tenant))
-                                .must(qb)
-                );
-            }
-
-            response = client.prepareSearch(indexes)
-                    .setRouting(tenant)
-                    .setSize(MAX_RESULT_LIMIT)
-                    .setVersion(true)
-                    .setQuery(bqb)
-                    .execute()
-                    .actionGet();
+            String response = elasticsearchRestHelper.fetch(
+                    ELASTICSEARCH_INDEX_NAME_READ, ELASTICSEARCH_DOCUMENT_TYPE, tenant, queries.get(0));
```

Comment:
It looks like the old code was using all the elements in the "queries" list, but here you only use the first.  why?

Reply:
I think, I may need to fix this. Earlier I saw that it's being called by search with query collection param and which itself is getting called by another search method with only one query parameter. Hence, I made that change. But, I just found out that it might be getting called from RollupHandler.getRollupByGranularity() method. So, I will dig into this further. Good catch.